### ISSUE TITLE
Label for Select and Slider (Form) - Provide a valid label for form fields - (2036570477)

### ISSUE THREAD

**Issue by dqateam** [Esri/calcite-components#5627]
### Module: 
 40 Form 

### Violation: 
 Provide a valid label for form fields

![image](https://user-images.githubusercontent.com/5023024/202567060-8abe9d2a-546e-444e-ab4e-7cc3b79054fc.png) 

### WCAG Reference: 
- [1.3.1 Info and Relationships](https://www.w3.org/WAI/WCAG21/Understanding/info-and-relationships.html)
- [3.3.2 Labels or Instructions](https://www.w3.org/WAI/WCAG21/Understanding/labels-or-instructions.html)
- [4.1.2 Name, Role, Value](https://www.w3.org/WAI/WCAG21/Understanding/name-role-value.html) 

### Instance ID: 
 2036570477 

### Severity: 
 10 

### Description: 
 [Issue]
There are form controls with visual labels that are not programmatically associated with the control. Examples include:

- Where did you hear about Calcite components?
- How many components will you be using?
- What is your favorite component(s)?

[User Impact]
Screen reader users will have difficulty determining the purpose of these controls. Speech input users will have difficulty navigating to them.

[Code Reference] 
 ```
<select class="select">
  <option label="Please selectâ¦" value="">Please selectâ¦</option>
  <option label="Esri DevSummit" value="">Esri DevSummit</option>
  <option label="Search engine" value="">Search engine</option>
  (...)
</select>
 ``` 

### Note: 
 [Suggestion]
Provide a valid label for form fields. The best way to do this is to use a ``` <label> ``` element with a for attribute set to the ID of the field it labels. For custom controls, use aria-labelledby on the control set to the ID of the labelling text.

[Compliant Code Example] 
 ```
<label for="dropdown-label">Where did you hear about Calcite components?</label>
<select id="dropdown-label" class="select">
  <option label="Please selectâ¦" value="">Please selectâ¦</option>
  <option label="Esri DevSummit" value="">Esri DevSummit</option>
  <option label="Search engine" value="">Search engine</option>
  (...)
</select>
 ```
 

### Media Type: 
 Forms 

### Additional Resources: 
- Report Source: Product accessibility evaluation conducted on the Calcite Design System in August, 2022.

**Comment by geospatialem** [comment]
`label` is not added to [the form](https://geospatialem.github.io/calcite-samples/0-form.html) for the following components:
- `select`
- `slider`
- `combobox`

![image](https://user-images.githubusercontent.com/5023024/202567060-8abe9d2a-546e-444e-ab4e-7cc3b79054fc.png)

**Comment by geospatialem** [comment]
@driskull This is confirmed for the `select` and `slider` components via [Codepen](https://codepen.io/geospatialem/pen/xxzWJLp) and the [testing page](https://geospatialem.github.io/calcite-samples/0-form.html) ([code](https://github.com/geospatialem/calcite-samples/blob/main/0-form.html)).

Added to the 1.0 milestone to address prior to release.

The JAWS transcript when working with the Codepen example:

```
Form elements and label: select, slider, combobox
heading level 2 label + select
Combo box 
Please select…
To change the selection use the Arrow keys.
A native select Combo box 
one
To change the selection use the Arrow keys.
left right slider 
0
To increase or decrease use the Arrow keys.
Native slider left right slider 
50
To increase or decrease use the Arrow keys.

```

**Comment by geospatialem** [comment]
Verified in `1.0.8-next.1` with JAWS and NVDA with the [following Codepen](https://codepen.io/geospatialem/pen/xxzWJLp).

Note that NVDA does not read the `label` for `slider`, however this is also the case with the native slider, so is likely a bug/discrepancy with NVDA and not CC.
